src/store/homework/slice.ts

```diff
@@ -0,0 +1,51 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { ProcessEnum } from 'utils/constants';
+import { HomeworkModel, HomeworkStatus } from 'api/homework/types';
+import { SerializedError } from 'api/core';
+import type { HomeworkState } from './types';
+
+const initialState: HomeworkState = {
+  homework: {
+    status: HomeworkStatus.Draft,
+    text: '',
+  },
+  process: ProcessEnum.Initial,
+  error: null,
+};
+
+const homeworkSlice = createSlice({
+  name: 'homework',
+  initialState,
+  reducers: {
+    resetHomeworkState: () => initialState,
+  },
+  extraReducers: (builder) => {
+    builder
+      .addMatcher(
+        ({ type }) => /^homework.*?\/fulfilled/.test(type as string),
+        (state, { payload }) => {
+          state.homework = payload as HomeworkModel;
+          state.process = ProcessEnum.Succeeded;
+          state.error = null;
+        }
```

Comment:
Для группировки санок в тулките есть специальные утилки, и они всё типизируют как нужно:

```suggestion
        isFulfilled(fetchHomeworkByLessonId, updateHomework),
        (state, { payload }) => {
          state.homework = payload;
          state.process = ProcessEnum.Succeeded;
          state.error = null;
        }
```